soroban-env-host/src/storage.rs

```diff
@@ -150,35 +154,40 @@ impl Storage {
         let ty = AccessType::ReadOnly;
         match self.mode {
             FootprintMode::Recording(ref src) => {
-                self.footprint.record_access(key, ty);
+                self.footprint.record_access(key, ty)?;
                 // In recording mode we treat the map as a cache
                 // that misses read-through to the underlying src.
-                if !self.map.contains_key(key) {
-                    self.map.insert(key.clone(), Some(src.get(key)?));
+                if !self.map.contains_key(key)? {
+                    self.map.insert(key.clone(), Some(src.get(key)?))?;
                 }
             }
             FootprintMode::Enforcing => {
                 self.footprint.enforce_access(key, ty)?;
             }
         };
-        match self.map.get(key) {
+        match self.map.get(key)? {
             None => Err(ScHostStorageErrorCode::MissingKeyInGet.into()),
             Some(None) => Err(ScHostStorageErrorCode::GetOnDeletedKey.into()),
             Some(Some(val)) => Ok(val.clone()),
         }
     }
 
+    pub fn metered_get(&mut self, host: &Host, key: &LedgerKey) -> Result<LedgerEntry, HostError> {
```

Comment:
As mentioned elsewhere I think metered_get here is redundant with the storage using MeteredMap inside itself.

Reply:
Fixed. Removed these "metered_" methods.